src/groupByKey.ts

```diff
@@ -2,6 +2,21 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<Item>(
+  items: Item[],
+  key: keyof Item,
+): GroupsMap<Item> {
+  const result: GroupsMap<Item> = {};
+
+  for (const item of items) {
+    const value = String(item[key]);
+
+    if (value in result) {
+      result[value].push(item);
+    } else {
+      result[value] = [item];
+    }
```

Comment:
To improve readability, you may consider using the logical OR assignment operator (||=) introduced in ES2021. This will allow you to simplify the conditional assignment to `result[value]`.